### PR TITLE
Add support for Max Guard and G-Max Moves to tooltips

### PR DESCRIPTION
**Pull request by HoeenCoder**

### CODE REVIEW

File: src/battle-tooltips.ts [hunk 1]
```diff
@@ -476,14 +479,9 @@ class BattleTooltips {
 				});
 				zEffect = this.getStatusZMoveEffect(move);
 			} else {
-				let moveid;
-				if (item.zMoveType) {
-					moveid = BattleTooltips.zMoveTable[item.zMoveType as TypeName];
-				} else {
-					moveid = BattleTooltips.maxMoveTable[move.type as TypeName];
-				}
+				let moveid = BattleTooltips.zMoveTable[item.zMoveType as TypeName];
```
Comment (KrisXV) [inline src/battle-tooltips.ts]: nitpick but this should be called `moveName` since the moves in the zMoveTable aren't IDs

**Comment by HoeenCoder** [comment]
Fixes #1408

**Comment by HoeenCoder** [comment]
Bump @Zarel

File: src/battle-dex-data.ts [hunk 2]
```diff
@@ -1177,7 +1177,7 @@ class Move implements Effect {
 				} else if (this.basePower >= 45) {
 					this.gmaxPower = 75;
 				} else  {
-					this.gmaxPower = 10;
+					this.gmaxPower = 70;
```
Comment (Zarel) [inline src/battle-dex-data.ts]: Does this need to be fixed on server, too, or is it implemented differently there?

Comment (Zarel) [reply src/battle-dex-data.ts]: Looks like SadisticMystic fixed it here, but also fixed some other things which should probably be propagated here.

https://github.com/smogon/pokemon-showdown/commit/b2e9336c327efbd22d36eb69f9b7905cd3955fb4

File: src/battle-tooltips.ts [hunk 3]
```diff
@@ -268,13 +268,16 @@ class BattleTooltips {
 		let buf: string;
 		switch (type) {
 		case 'move':
-		case 'zmove': { // move|MOVE|ACTIVEPOKEMON
+		case 'zmove':
+		case 'maxmove': { // move|MOVE|ACTIVEPOKEMON
```
Comment (Zarel) [inline src/battle-tooltips.ts]:
```suggestion
		case 'maxmove': { // move|MOVE|ACTIVEPOKEMON|[GMAXMOVE]
```

**Comment by HoeenCoder** [comment]
I have not forgotten about this, I'll finish it up when I get off LOA in a few days/get a bit of time to do it.

**Comment by Zarel** [comment]
Thanks!

**Pull request merged**